### DISCUSSION TITLE
Why is it better to replace Newtonsoft.Json

### DISCUSSION THREAD

**Discussion by ddikhtyar** [ElectronNET/Electron.NET#818] [General]
Newtonsoft.Json puts a limit on the popularity of the entire Electron.NET because many corporations have included this package in the banned list.

**Comment by liebki** [comment]
@ddikhtyar why is it on ban lists?

**Reply by ddikhtyar** [reply]
because the vulnerabilities prevent the security department from allowing its use.

**Comment by Denny09310** [comment]
If @FlorianRappl and @GregorBiswanger are interested, since they're currently testing the alpha with the new architecture, I can try and work on replacing all Newtonsoft.Json usages with System.Text.Json, assuming there's no specific reason to avoid making this change.

**Reply by FlorianRappl** [reply]
That would be super useful! Yes please!

**Reply by softworkz** [reply]
@Denny09310 - Please note that the new branch uses a newer version of SocketIO with integreated serialization. You'll also need to change the `SocketIO.Serializer.NewtonsoftJson` nuget to [SocketIO.Serializer.SystemTextJson](https://www.nuget.org/packages/SocketIO.Serializer.SystemTextJson) in order to avoid having mixed serializer implementations.

**Reply by Denny09310** [reply]
Thanks for the guidance. I’ve created a feature [branch](https://github.com/Denny09310/Electron.NET/tree/feature/system-text-json) with the updated code. However, I'm not sure how to properly test it. Some things don’t seem to be working in the demos, in the develop branch, or in my branch, so I’m not confident about the results. Any advice on how to test this reliably?

**Reply by FlorianRappl** [reply]
I mean if something is not working we need to make it working -> so, sure it could be either that the demo is broken already or that it's something with the change. In any case, debugging / finding the root cause would be helpful.

In general using the demo application is the way to go. I think mid-term we should invest in some automation here.

**Reply by softworkz** [reply]
> Any advice on how to test this reliably?

Yes, use tihis: https://github.com/ElectronNET/Electron.NET/pull/913 😄

**Reply by Denny09310** [reply]
Thanks a lot!! 😄

**Reply by softworkz** [reply]
I stupidly misestimated the time required for this. I got the tests generated rather quickly as expected, but I hadn't considered that the tests would also uncover lots of issues that need fixing, so it took a few hours..

The tests do not provide full coverage, but I think they should be sufficient for uncovering any issues with the migration of serialization.

I've also created another PR for whitespace changes because I had seen your commits had mixed white-stpace/formatting changes with actuak´code changes - which makes it more difficult to see the actual code changes. Once that is merged and you rebase your (then-final) serialization branch, the whitespace changes will disappear from your commits (because these changes were already made).

**Reply by softworkz** [reply]
Did you dellete your question about 

> "session_clear_cache_and_storage_completes" is taking very long?

(got it by e-mail only) 

So just in case, the answer would be "no". I ran them on Windows only so far. These are my results:

<img width="747" height="192" alt="image" src="https://github.com/user-attachments/assets/4cc3b73e-486d-4426-aac2-6e8bf5693abf" />

The two failing tests are intentional (need fixing)

**Reply by Denny09310** [reply]
Yes, I was trying to see if it was a "my-problem" or a "test-problem" 😅. Mhm very strange then, I've only checked out to your branch without changing anything

**Reply by softworkz** [reply]
When you have to interrupt any test (whether by stop debugging or in Test Explorer, you'll likely have orphaned electron processes.

This will help: `taskkill /f /im electron.exe`

**Reply by softworkz** [reply]
Hm, and all other tests are running fine?
You're on Windows? Visual Studio?
Did you "Rebuild solution" (creates errors), then "Build Solution "(no errors)?

**Reply by softworkz** [reply]
If that doesn't help, we'll need to find out where it hangs.. Debug and use breakpoints rather than stepping:

<img width="1398" height="214" alt="image" src="https://github.com/user-attachments/assets/e7f2a90f-dfd9-47f9-b652-8f35c69c3665" />

**Reply by Denny09310** [reply]
Yes all good.

>This will help: `taskkill /f /im electron.exe`

That was the problem, I didn't noticed orphaned processes. Thanks again

**Reply by Denny09310** [reply]
I was able to integrate your tests in my branch and migrate to System.Text.Json, all but one test (except for the two already failing) are passing 😁

**Reply by softworkz** [reply]
Now that the test provides a fixed pattern, you can run Wireshark on the local loopback interface and execute the failing test once with and once without JSON migration.
Then you can compare the payloads which go over the wire.

I had looked for but not found a public interception point in SocketIO, otherwise I had created unit tests as well (without Electron running) - so Wireshark is probably the best bet.
